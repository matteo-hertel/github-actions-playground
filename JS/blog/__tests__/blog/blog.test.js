require('dotenv').config({
    path: `${__dirname}/../../.env`
});
jasmine.DEFAULT_TIMEOUT_INTERVAL = 10000;
const blog = require(`${__dirname}/../../modules/blog`);

describe("Blog Module", () => {
    it('should return a storable payload for a given file', () => {
        return blog.getStorable("test/2017-01-27_test-post-seo.md")
        .then((data) => {
            expect(data.name).toBeDefined();
            expect(data.seo).toBeDefined();
            expect(data.body).toBeDefined();
            expect(data.createdDate).toBeDefined();
            expect(data.updatedDate).toBeDefined();
        });
    });
});
