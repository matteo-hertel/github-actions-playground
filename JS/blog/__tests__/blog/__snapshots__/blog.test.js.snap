exports[`Blog Module should return a list of storable payloads on re-index 1`] = `
Array [
  Object {
    "body": "<h1>Test Post</h1>

<p>the copy of the test post will go here</p>",
    "createdDate": 1485460800,
    "name": "Test Post Seo",
    "seo": "2017-01-27-test-post-seo",
    "updatedDate": 1487369355,
  },
  Object {
    "body": "<h1>Test Post</h1>

<p>the copy of the test post will go here</p>",
    "createdDate": 1485547200,
    "name": "Test Post2 Seo",
    "seo": "2017-01-28-test-post2-seo",
    "updatedDate": 1487369355,
  },
  Object {
    "body": "<h1>Test Webhook</h1>

<p>the copy of the test post will go here for testing the webhook</p>",
    "createdDate": 1487016000,
    "name": "Test Post2 Seo",
    "seo": "2017-02-14-test-post2-seo",
    "updatedDate": 1487369355,
  },
]
`;

exports[`Blog Module should return a storable payload for a given file 1`] = `
Object {
  "body": "<h1>Test Post</h1>

<p>the copy of the test post will go here</p>",
  "createdDate": 1485460800,
  "name": "Test Post Seo",
  "seo": "2017-01-27-test-post-seo",
  "updatedDate": 1487369361,
}
`;
